.github/workflows/release.yml

```diff
@@ -7,27 +7,36 @@ on:
         type: choice
         description: Select the version change
         required: true
-        default: 'minor'
+        default: "minor"
         options:
           - major
           - minor
           - patch
 
+permissions:
+  contents: read
+
 jobs:
   release:
     runs-on: ubuntu-latest
     env:
       VERSION_CHANGE: ${{ github.event.inputs.versionChange }}
+
+    permissions:
+      contents: write
+      id-token: write
+
     steps:
       - name: Checkout 🛎️
-        uses: actions/checkout@v4
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
```

Comment:
These are now pinned to the latest full-length commit SHAs. This means that if `actions/checkout` is compromised (unlikely, but some of the third-party ones might), then you'll remain on the correct version. This comes with the downside of needing to manually bump them.

Reply:
But dependabot will let us know once a new version is available, so that we can update the version manually?

Reply:
Correct. Dependabot will open a PR, and you can check that it works.
It's helpful to pin even with GH actions b/c they're buggy sometimes, which is primarily why I do it. I'll usually just scan the release notes and the open issues and press merge for gh's. For third-party ones, I'll often skim the diff, but I've never run into issues security-wise. Nonetheless, I (and [GitHub](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions)) still recommend pinning just in case.

Reply:
Thanks for sharing that knowledge! It sounds good to me, let's try it out :)